### PR TITLE
Removing terminal state for Pendulum-v0 env

### PR DESCRIPTION
**Pull request by nishantkr18**
I see that we terminate the episode when we find the pendulum's angle within a threshold given by:

`state.Theta() > M_PI - angleThreshold ||
        state.Theta() < -M_PI + angleThreshold`

But doing this makes the environment very easy to train on. With **random actions**, the environment terminates after not more than 30 steps, as soon as it gets the pendulum within the threshold. While in the gym's implementation (https://gym.openai.com/envs/Pendulum-v0/), it's said that "goal is to swing it up so it stays upright.". 
I guess removing the threshold condition in `isTerminal` will do our job, we would then receive higher rewards for making the pendulum stay upright, and not just making it swing once :)

### CODE REVIEW

**Comment by zoq** [comment]
Nice catch! Looking at: https://github.com/openai/gym/wiki/Pendulum-v0#episode-termination do you think we can instead add a max steps parameter + check, `0` could mean no limit, but I guess some limit makes sense. Let me know what you think.

**Comment by nishantkr18** [comment]
> Nice catch! Looking at: https://github.com/openai/gym/wiki/Pendulum-v0#episode-termination do you think we can instead add a max steps parameter + check, `0` could mean no limit, but I guess some limit makes sense. Let me know what you think.

I think there are two ways we can achieve this. 

First is that we set `StepLimit` as 200 in TrainingConfig, and then in the implementation of the agent, it will be checked like here: https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/reinforcement_learning/q_learning_impl.hpp#L244

 Second, we can set `maxSteps` as 200 by default [here](https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp#L108) and then it will be checked in the environment implementation itself https://github.com/mlpack/mlpack/blob/master/src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp#L219

Personally I think the second option is better because it does the checking in the environment itself. Let me know what u think :)

**Comment by zoq** [comment]
> Personally I think the second option is better because it does the checking in the environment itself. Let me know what u think :)

Agreed, thanks for putting this together.

File: src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ class Pendulum
            const double dt = 0.05,
            const double angleThreshold = M_PI / 12,
            const double doneReward = 0.0,
-           const size_t maxSteps = 0) :
+           const size_t maxSteps = 200) :
```
Comment (zoq) [inline src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp:108]: Can you add a comment to the `maxStep` parameter above: "(Default: 200 steps)."?

**Comment by zoq** [comment]
Also, can you update the `HISTORY.md` file as well?

File: src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp (hunk 1 above)
Comment (nishantkr18) [reply src/mlpack/methods/reinforcement_learning/environment/pendulum.hpp:108]: sorry I forgot.. done now :)

**Comment by nishantkr18** [comment]
> Also, can you update the `HISTORY.md` file as well?

Done :+1:

**Comment by zoq** [comment]
Nice, let me fix the merge conflict real quick.

**Comment by zoq** [comment]
Okay, let's see if the build matrix comes back green.

**Comment by nishantkr18** [comment]
> Nice, let me fix the merge conflict real quick.

Oh.. didn't see that coming😅

**Comment by favre49** [comment]
I think we can merge this now. Thanks for the contribution!

**Pull request merged**